Compiling with disable_3d and disable_advanced_gui and tools=yes does not work

Building with VisualStudio 2010 using scons with the following command lines has numerous unresolved externals:

scons platform=windows disable_3d=yes

(Using disable_advanced_gui=yes with the above also fails but I'm not sure if using this adds more unresolved externals)

Build error:
bin\godot.windows.tools.exe : fatal error LNK1120: 114 unresolved externals
scons: **\* [bin\godot.windows.tools.exe] Error 1120
scons: building terminated because of errors.

Godot 3 still can't be compiled with disable_3d=yes and disable_advanced_gui=yes.
```
bin\godot.windows.opt.debug.64.mono.exe : fatal error LNK1120: 8  unresolved externals
scons: *** [bin\godot.windows.opt.debug.64.mono.exe] Error 1120
scons: building terminated because of errors.
```

I'm trying to compile Godot using the following command from Windows 10 (latest version from git):
`scons -j6 platform=android --config=force target=release tools=no use_lto=yes deprecated=no xml=no disable_3d=yes disable_advanced_gui=yes builtin_bullet=no builtin_enet=no builtin_libogg=no builtin_libtheora=no builtin_libvorbis=no builtin_libvpx=no builtin_libwebsockets=no builtin_mbedtls=no builtin_miniupnpc=no builtin_opus=no builtin_recast=no builtin_squish=no builtin_thekla_atlas=no module_bmp_enabled=no module_bullet_enabled=no module_csg_enabled=no module_dds_enabled=no module_enet_enabled=no module_etc_enabled=no module_gdnative_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_jpg_enabled=no module_mbedtls_enabled=no module_mobile_vr_enabled=no module_ogg_enabled=no module_opus_enabled=no module_pvr_enabled=no module_recast_enabled=no module_squish_enabled=no module_stb_vorbis_enabled=no module_svg_enabled=no module_tga_enabled=no module_thekla_unwrap_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_upnp_enabled=no module_visual_script_enabled=no module_vorbis_enabled=no module_webm_enabled=no module_websocket_enabled=no`

It compile everything but the linker give these errors:
```
[ 83%] Linking Static Library ==> modules\freetype\libfreetype_builtin.android.opt.armv7.neon.a
[ 83%] Linking Static Library ==> core\libcore.android.opt.armv7.neon.a
[ 83%] Linking Shared Library ==> bin\libgodot.android.opt.armv7.neon.so
[ 83%] =====
D:\Android\sdk\ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/windows-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin\ld: warning: ignoring --threads: D:\Android\sdk\ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/windows-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin\ld was compiled without thread support
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_stop_simulation(Node*): error: undefined reference to 'PhysicalBone::set_simulate_physics(bool)'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_stop_simulation(Node*): error: undefined reference to 'PhysicalBone::set_static_body(bool)'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_stop_simulation(Node*): error: undefined reference to 'typeinfo for PhysicalBone'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_start_simulation(Skeleton const*, Node*, Vector<int> const&): error: undefined reference to 'PhysicalBone::set_simulate_physics(bool)'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_start_simulation(Skeleton const*, Node*, Vector<int> const&): error: undefined reference to 'PhysicalBone::set_simulate_physics(bool)'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_start_simulation(Skeleton const*, Node*, Vector<int> const&): error: undefined reference to 'PhysicalBone::set_static_body(bool)'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _pb_start_simulation(Skeleton const*, Node*, Vector<int> const&): error: undefined reference to 'typeinfo for PhysicalBone'
scene\libscene.android.opt.armv7.neon.a(skeleton.android.opt.armv7.neon.o):scene\3d\skeleton.cpp:function _physical_bones_add_remove_collision_exception(bool, Node*, RID): error: undefined reference to 'typeinfo for CollisionObject'
clang++.exe: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/SConstruct
+++ b/SConstruct
@@ -417,11 +417,19 @@ if selected_platform in platform_list:
     if env['tools']:
         env.Append(CPPDEFINES=['TOOLS_ENABLED'])
     if env['disable_3d']:
-        env.Append(CPPDEFINES=['_3D_DISABLED'])
+        if env['tools']:
+            print("Build option 'disable_3d=yes' cannot be used with 'tools=yes' (editor), only with 'tools=no' (export template).")
+            sys.exit(255)
+        else:
+            env.Append(CPPDEFINES=['_3D_DISABLED'])
     if env['gdscript']:
         env.Append(CPPDEFINES=['GDSCRIPT_ENABLED'])
     if env['disable_advanced_gui']:
-        env.Append(CPPDEFINES=['ADVANCED_GUI_DISABLED'])
+        if env['tools']:
+            print("Build option 'disable_advanced_gui=yes' cannot be used with 'tools=yes' (editor), only with 'tools=no' (export template).")
+            sys.exit(255)
+        else:
+            env.Append(CPPDEFINES=['ADVANCED_GUI_DISABLED'])
     if env['minizip']:
         env.Append(CPPDEFINES=['MINIZIP_ENABLED'])
     if env['xml']:

